app/main.py

```diff
@@ -1,2 +1,69 @@
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.nodes = [[]] * 8
+
+    def __setitem__(self, key: any, value: any) -> None:
+
+        if self.__len__() >= self.capacity * 2 / 3:
+            self.capacity *= 2
+            old_nodes = self.nodes
+            self.nodes = [[]] * self.capacity
+
+            for node in old_nodes:
+                if node:
+                    if not self.nodes[node[2] % self.capacity]:
+                        self.nodes[node[2] % self.capacity] = node
+                    else:
+                        for i in range(0, self.capacity):
+                            if not self.nodes[i]:
+                                self.nodes[i] = node
+                                break
+
+        node = self.nodes[hash(key) % self.capacity]
+        if node:
+            if not node[0] == key:
+                for i in range(0, self.capacity):
+                    if not self.nodes[i] or self.nodes[i][0] == key:
+                        self.nodes[i] = [key, value, hash(key)]
+                        break
+            else:
+                self.nodes[hash(key) % self.capacity] = [key, value, hash(key)]
+        else:
+            self.nodes[hash(key) % self.capacity] = [key, value, hash(key)]
+
+    def __getitem__(self, key: any) -> any:
+        node = self.nodes[hash(key) % self.capacity]
```

Comment:
When searching for a key in the hash table, you're using:

```python
node = self.nodes[hash(key) % self.capacity]
```

This assumes the key is always at the hashed index, but it won't work if there are collisions (multiple keys hashed to the same index). You need to handle collisions by iterating through the list of items at that index to find the correct key.